### PR TITLE
Consider roles instead of raw permission classes

### PR DESCRIPTION
**Pull request by lubosmj**
[noissue]

### CODE REVIEW

**Comment by lubosmj** [comment]
The changes address a bug in `PushCrossRepoBlobMountTestCase` where model and object permissions are no longer used after merging the roles PR.

File: pulp_container/tests/functional/api/test_rbac_repo_versions.py [hunk 1]
```diff
@@ -271,17 +271,18 @@ def setUpClass(cls):
         cls.distribution = cls.distributions_api.list(name="test-1").results[0]
 
         cls.user_pull = gen_user(
-            object_permissions=[
-                ("container.pull_containerdistribution", cls.distribution.pulp_href)
+            object_roles=[
+                ("container.containerdistribution_consumer", cls.distribution.pulp_href),
+                ("container.containernamespace_consumer", cls.distribution.namespace),
```
Comment (mdellweg) [inline pulp_container/tests/functional/api/test_rbac_repo_versions.py:276]: Would not giving them the namespace role be sufficient? (I think it i should be.)

File: pulp_container/tests/functional/api/test_rbac_repo_versions.py [hunk 2]
```diff
@@ -271,17 +271,18 @@ def setUpClass(cls):
...
             ]
         )
         cls.user_push = gen_user(
-            [
-                "container.push_containerdistribution",
-                "container.namespace_push_containerdistribution",
+            object_roles=[
+                ("container.containerdistribution_collaborator", cls.distribution.pulp_href),
+                ("container.containernamespace_collaborator", cls.distribution.namespace),
```
Comment (mdellweg) [inline pulp_container/tests/functional/api/test_rbac_repo_versions.py:282]: Again, i think one of them should be sufficient.

File: pulp_container/tests/functional/utils.py [hunk 3]
```diff
@@ -57,14 +57,8 @@
 TOKEN_AUTH_DISABLED = utils.get_pulp_setting(cli_client, "TOKEN_AUTH_DISABLED")
 
 
-def gen_user(model_permissions=None, object_permissions=None, model_roles=None, object_roles=None):
```
Comment (mdellweg) [inline pulp_container/tests/functional/utils.py:60]: :+1:

File: pulp_container/tests/functional/api/test_rbac_repo_versions.py (hunk 2 above)
Comment (lubosmj) [reply pulp_container/tests/functional/api/test_rbac_repo_versions.py:282]: Yes, you are right. I tested the push permission with the namespace permission and it was working.

Comment (mdellweg) [reply pulp_container/tests/functional/api/test_rbac_repo_versions.py]: Now i'm asking myself whether we should have a test for both individually.

Comment (lubosmj) [reply pulp_container/tests/functional/api/test_rbac_repo_versions.py]: The test case tries to verify whether a user has all the required permissions to perform blob mounting (`pull` for one repository & `push` for another repository). We are not trying to test the whole access_policy workflow for pushing content. I am fine with namespace permissions only.

Comment (lubosmj) [reply pulp_container/tests/functional/api/test_rbac_repo_versions.py]: Maybe, I should move the test case to another file. The name evokes that we are trying to test the whole RBAC suite, which is wrong.

Comment (mdellweg) [reply pulp_container/tests/functional/api/test_rbac_repo_versions.py]: Well, we are creating users and so we testing for rbac features. I think, the name fits.

**Review by mdellweg — APPROVED** [review]

**Pull request merged**